scapy/modules/packet_viewer/row_formatter.py

```diff
@@ -0,0 +1,186 @@
+
+        # Do not trim last column. Usually it's the data column
+        # so allow it to be as long as necessary
+        name = self.columns[-1][0]
+        format_string += "{" + name + "}"
+        return format_string
+
+    def get_all_columns(self):
+        # type: () -> List[Tuple[str, int, Callable[[Packet], str]]]  # noqa: E501
+        """
+        Depending if a basecls filter is configured, this function returns
+        either a standard column configuration which uses the packets repr
+        function, or a custom column configuration based on the basecls
+        :return: A default or a basecls specific column configuration
+        """
+        if self.basecls is None:
+            return self.get_default_columns() + [("REPR", 50, Packet.__repr__)]
+        else:
+            return self.get_default_columns() + self.get_config_columns() + \
+                self.fields_to_columns()
+
+    def get_default_columns(self):
+        # type: () -> List[Tuple[str, int, Callable[[Packet], str]]]
+        """
+        Return the default column configuration
+        :return: The default column configuration
+        """
+        nr_messages = count()
+        # TODO: Check if _id_map can be refactored
+        return [
+            ("NO", 5, lambda p: str(
+                self._id_map.setdefault(p.time, next(nr_messages)))),
+            ("TIME", 11, self.relative_time)
+        ]
+
+    def get_config_columns(self):
+        # type: () -> List[Tuple[str, int, Callable[[Packet], str]]]
+        """
+        Return all columns from Scapys configuration dependent on the
+        basecls
+        :return: A columns configuration from
+                 conf.contribs["packet_viewer_columns"] for the current basecls
+                 if a configuration is present.
+        """
+        try:
+            return conf.contribs["packet_viewer_columns"].get(self.basecls)
+        except KeyError:
+            return []
+
+    def fields_to_columns(self, width=12):
+        # type: (int) -> List[Tuple[str, int, Callable[[Any], str]]]
+        """
+        Returns a column configuration automatically deduced by the configured
+        basecls. All fields of this Packet_metaclass will be returned
+        :param width: The width of a field
+        :return: A automatically generated column configuration
+        """
+        columns = []  # type: List[Tuple[str, int, Callable[[Any], str]]]
+        if self.basecls is None:
+            return columns
+
+        for field_desc in self.basecls.fields_desc:
+            # Instantiate a value for the field to check its type
+            dummy_field_val = self.basecls().getfieldval(field_desc.name)
+            # If byte, python adds quotation marks to the repr
+            # bytes_to_repr removes it
+            # We use repr() over str() because otherwise byte values
+            # like 0x0a ('\n') would change the layout
+            if isinstance(dummy_field_val, bytes):
```

Comment:
```suggestion
            callback = lambda p: self.text_to_repr(p, field_desc.name) if isinstance(dummy_field_val, bytes) else lambda p: self.field_to_repr(p, field_desc.name)
```
?

Reply:
I think the current version is more readable